packages/rrweb/src/record/error-handler.ts

```diff
@@ -24,6 +27,13 @@ export const callbackWrapper = <T extends Callback>(cb: T): T => {
     try {
       return cb(...rest);
     } catch (error) {
+      // Some libraries (styled-components) rely on an exception to be thrown.
+      // Attach an optional property here to allow `errorHandler` to make
+      // additional decisions (e.g. to re-throw).
+      if (errorSource) {
+        error.__source__ = errorSource;
+      }
```

Comment:
Since this only happens if we have `errorHandler` defined, this probably only affects us. I opted to make this generic so that `errorHandler` be the source of truth vs attaching a meta property like `__allowThrow__`